### PR TITLE
GH-1116 Highlight Update Targets

### PR DESCRIPTION
**Pull request by GordonSmith**
HTML Marshaller will highlight which widgets update other widgets.
Fixed bug in Grid.getCell (broken when cell density was introduced).
Partial fix for performance issue (opening new issue(s) for my thoughts).

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage Please Review

**Comment by mlzummo** [comment]
@GordonSmith do you have a link to show this working?

File: src/layout/Grid.js [hunk 1]
```diff
@@ -87,7 +87,9 @@
         return this;
     };
 
-    Grid.prototype.getCell = function (row, col) {
+    Grid.prototype.getCell = function (_row, _col) {
+        var row = _row * this.cellDensity();
+        var col = _col * this.cellDensity();
```
Comment (jbrundage) [inline src/layout/Grid.js]: @GordonSmith This might cause undesired effects... this assumes that _row and _col aren't already adjusted for cellDensity.

Comment (jbrundage) [inline src/layout/Grid.js]: ...but everything seems to work normally

Comment (jbrundage) [inline src/layout/Grid.js]: @GordonSmith Actually this does break the ability to drop a widget ontop of another widget to swap their locations in designMode.

When the cellDensity is changed it is automatically applied to each cell's gridRow, gridCol, gridRowSpan and gridColSpan. Lines 91 and 92 shouldn't be necessary.

Comment (jbrundage) [inline src/layout/Grid.js]: ...it also prevents the user from being able to "select" a cell

Comment (GordonSmith) [inline src/layout/Grid.js]: That means that setContent(5, 5, widget) is not symmetrical with a getCell(5, 5)?

Comment (jbrundage) [inline src/layout/Grid.js]: Valid point, yes I should have considered getCell might be used outside of the layout/Grid file.

I suppose a new method **getDensityCell()** should be created/used inside the layout/Grid file and getCell should be as you have it in this PR.

Comment (GordonSmith) [inline src/layout/Grid.js]: For now I will add a getWidgetCell = function(Widget) method.

**Comment by GordonSmith** [comment]
http://rawgit.com/GordonSmith/Visualization/GH-1116/demos/dermatology.html?src/marshaller/HTML

@jbrundage Changes made.

**Comment by GordonSmith** [comment]
REbased @jbrundage Please Review

**Comment by mlzummo** [comment]
looks good

**Pull request merged**